File: responses/H3G23A-8H4/1_sum_of_integers_in_a_string.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(a)
  return 0 if a.length == 0
  hash_table = {}
  a.each do |element|
    if hash_table[element]
      hash_table[element] += 1
    else
      hash_table[element] = 1
    end
  end
  hash_table.values.max
end

File: responses/ZJBAVP-GPH/1_sum_of_integers_in_a_string.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(A) {
    var max;
    var count = 0;
    
    for (var i=0; i<A.length; i++) {
        if (max === undefined) {
            max = A[i];
        } else {
            max = Math.max(max, A[i]);
        }
    }
    
    for (var j=0; j<A.length; j++) {
        if (A[j] === max) {
            count++;
        }
    }
    
    return count;
}